src/app/content/studyGuides/hooks/updateQueryWithSummaryFiltersHookBody.ts

```diff
@@ -0,0 +1,27 @@
+import { getType } from 'typesafe-actions';
+import { replace } from '../../../navigation/actions';
+import * as navigation from '../../../navigation/selectors';
+import { updateQuery } from '../../../navigation/utils';
+import { ActionHookBody } from '../../../types';
+import updateSummaryFilters from '../../highlights/utils/updateSummaryFilters';
+import * as actions from '../actions';
+import { summaryFilters } from '../selectors';
+
+export const hookBody: ActionHookBody<
+  typeof actions.setDefaultSummaryFilters |
+  typeof actions.updateSummaryFilters
+> = (services) => (action) => {
+  const state = services.getState();
+  const existingQuery = navigation.query(state);
+  const match = navigation.match(state);
+  const filters = summaryFilters(state);
+
+  if (!match) { return; }
+
+  const updatedFilters = action.type === getType(actions.setDefaultSummaryFilters)
+    ? action.payload
+    : updateSummaryFilters(filters, action.payload);
+  services.dispatch(replace(match, {
+    search: updateQuery(updatedFilters as any as Record<string, string[]>, existingQuery),
+  }));
+};
```

Comment:
I've replaced the setDefaultSummaryFilters action with a query update but I'm not sure the best way to do the same for updateSummaryFilters because of shared logic with highlights, eg the ColorFilter component. We can take a look if you want.